0000-settlement-engine/0000-settlement-engine.md

```diff
@@ -0,0 +1,353 @@
+
+#### Scale conversions
+
+Since the systems MAY use different scales, when interfacing with one another, care must be taken to ensure the system with additional precision tracks the amount leftover by the other system so it may accumulate and eventually be correctly accounted for.
+
+If one of the systems receives a request with a **[`Quantity`](#quantity-type)** denominated in a unit more precise than its unit, it SHOULD convert the quantity into its native unit. If so, the resulting amount MUST be rounded down before fulfilling the request.
+
+### Settlement Engine HTTP API
+
+#### Initiate an account
+
+Inform the settlement engine that a new accounting relationship was instantiated. The settlement engine MAY perform tasks as prerequisite to settle the new account. For example, a settlement engine implementation might send messages to the peer to exchange ledger identifiers or to negotiate terms of the settlement arrangement.
+
+The transport MUST be the only entity invoking this instruction to ensure the account identifier can be correlated with the correct peer to send and receive messages. Note that the transport also MUST have a mechanism to notify the accounting system that a new account should be instantiated (out-of-scope of this specification).
+
+##### Request
+
+```http
+POST /accounts/:id HTTP/1.1
+```
+
+##### Response
+
+```http
+HTTP/1.1 201 CREATED
+```
+
+#### Perform outgoing settlement
+
+Reconcile a liability owed to the peer for the given amount.
+
+Note that the settlement engine MAY accrue owed settlements without settling immediately due to varying arrangements, implementations, or conditions. The settlement engine SHOULD persist amounts owed to the peer that have yet to be settled so they can be settled later.
+
+The accounting system MUST be the only entity invoking this instruction and MUST preemptively debit this amount from the accounts payable, or some liability account tracking a balance owed to the peer, to ensure accurate accounting.
+
+##### Request
+
+```http
+POST /accounts/:id/settle HTTP/1.1
+Accept: application/json
+Content-Type: application/json
+```
+
+> **[`Quantity`](#quantity-type)** to settle
+>
+> - If the amount is `0`, the settlement engine MAY retry failed or queued settlements.
+
+##### Response
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json
+```
+
+> **[`Quantity`](#quantity-type)** enqueued to settle
+>
+> - This response does not guarantee that a settlement was executed.
+> - If the settlement engine uses a unit less precise than the accounting system's unit, in order to prevent the two systems getting out-of-sync, it must indicate the amount it queued so the accounting system can correctly track the amount leftover.
+
+#### Handle incoming request
+
+Respond to an incoming message from the given peer's settlement engine. The transport MUST be the only entity invoking this instruction.
+
+##### Request
+
+```http
+POST /accounts/:id/handleMessage HTTP/1.1
```

Comment:
I think `handle` can be implicit here given you're sending a message to be handled by the server. What about `/accounts/:id/message`?

Reply:
Addressed here, changed to `messages`: https://github.com/interledger/rfcs/compare/4e742ae54dcd738b67a4a541f9070837a3c16e98...ko-settlement-engine#diff-990a915317f4e35578edfa70631bd07dR294